### ISSUE TITLE
Strict type checking like `if type(x) is T:` leads to wrong type restrictions

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#572]
## Environment data

-   Language Server version: Pylance 2020.11.0
-   OS and version: Window 7
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.8.5

## Actual Behavior

I have written a function to cast instances of any sub-classes of `datetime.date` as direct instances of `datetime.date`, so they can be used as SQL parameters.

```python
import datetime as dt

def cast_as_builtin_date(date: dt.date) -> dt.date:
    if type(date) is dt.date:
        return date

    # Pylance says the type of `date` from now on will be 'Never',
    # but it still can necessarily be any sub-class of `datetime.date`
    # except for `datetime.date` itself.
    return dt.date(date.toordinal())
```

The type of argument `date` after the type check is determined as "Never" by Pylance, and as a result, the call of `date.toordinal()` is regarded problematic. It seems that Pylance treats `if type(x) is T:`  just the same as `if isinstance(x, T):`, although they are different in some degrees.

## Code Snippet / Additional Information

As a side note, code like `if type(x) == T` and `if T is type(x)` works just fine and can be used as a work-around.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.11.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#2020111-11-november-2020
